src/lib/Liblog/pbs_log.c

```diff
@@ -867,7 +867,7 @@ log_record(int eventtype, int objclass, int sev, const char *objname, const char
 #endif
 	int    rc = 0;
 	FILE  *savlog;
-	static char slogbuf[LOG_BUF_SIZE];
+	char slogbuf[LOG_BUF_SIZE];
```

Comment:
I believe there is a performance advantage to keeping this as a static so that it doesn't have to be reallocated on the stack each time log_record is called.

Reply:
I had given some thought to that and researched online, and according to what I found, it seemed like the performance hit would be negligible. Here's a thread which talks about this:
https://stackoverflow.com/questions/3730000/can-static-local-variables-cut-down-on-memory-allocation-time

So, I'd like to keep this change, otherwise we'd have to create a thread-safe version of log_record() and have the scheduler use that, which doesn't seem worth it. But please let me know if you'd like me to spend more time on this.

Reply:
No need to spend more time on this. Thanks for the analysis and response, I'm OK with leaving this as you have it.